crates/aranya-client-capi/src/api/defs.rs

```
@@ -928,3 +988,95 @@ pub unsafe fn afc_recv_data(
 
     Ok(true)
 }
+
+// TODO: query_devices_on_team
+
+// TODO: query_device_role
+
+/// Query device's keybundle.
+///
+/// @param client the Aranya Client [`Client`].
+/// @param team the team's ID [`TeamId`].
+/// @param device the device's ID [`DeviceId`].
+/// @param __output the device's key bundle [`KeyBundle`].
+///
+/// @relates AranyaClient.
+pub unsafe fn query_device_keybundle(
+    client: &mut Client,
+    team: &TeamId,
+    device: &DeviceId,
+) -> Result<KeyBundle, imp::Error> {
+    let client = client.deref_mut();
+    let keys = client
+        .rt
+        .block_on(client.inner.team(team.0).query_device_keybundle(device.0))?;
+    Ok(KeyBundle::from_underlying(keys))
+}
+
+// TODO: query_device_label_assignments
+
+/// Query device's AFC network identifier.
+///
+/// @param client the Aranya Client [`Client`].
+/// @param team the team's ID [`TeamId`].
+/// @param device the device's ID [`DeviceId`].
+/// @param network identifier string [`NetIdentifier`].
+///
+/// @relates AranyaClient.
+pub unsafe fn query_afc_net_identifier(
+    client: &mut Client,
+    team: &TeamId,
+    device: &DeviceId,
+    ident: &mut [MaybeUninit<c_char>],
+) -> Result<(), imp::Error> {
+    let client = client.deref_mut();
+    let net_identifier = client
+        .rt
+        .block_on(client.inner.team(team.0).query_afc_net_identifier(device.0))?;
+    let mut ident_len = net_identifier.0.len();
```

This is throwing away then length instead of returning it to the caller. You want this:

```rust
pub unsafe fn query_afc_net_identifier(..., ident: &mut MaybeUninit<c_char>], ident_len: &mut usize) -> Result<(), imp::Error> {
    [...]
    let net_identifier = ...;
    let ident = aranya_capi_core::try_as_mut_slice!(ident, ident_len);
    aranya_capi_core::write_c_str(ident, &net_identifier, ident_len)?;
}
```

---

Thank you, pushed a commit for this